sdk/core/azure-core/src/main/java/com/azure/core/util/UrlBuilder.java

```diff
@@ -291,6 +294,14 @@ public String toString() {
         return result.toString();
     }
 
+    /**
+     * Returns the map of parsed URLs and their {@link UrlBuilder UrlBuilders}
+     * @return the map of parsed URLs and their {@link UrlBuilder UrlBuilders}
+     */
+    static Map<String, UrlBuilder> getParsedUrls() {
+        return PARSED_URLS;
+    }
```

Comment:
just leaving a comment as a reminder for the future when we improve it - We noted that for 98% of the cases, the path will be relative, and the `RestProxy` will not use the cached or cloned `UrlBuilder`, so we could potentially optimize it by not allocating the `UrlBuilder` for those cases, but rather storing a flag indicating that it's parsed and has no sehema. But this requires detailed discussion and testing, so let's keep this as a note for the future.